trace/trace.go

```diff
@@ -319,12 +313,40 @@ func isTraceStateKeyValueValid(kv attribute.KeyValue) bool {
 		valueFormatRegExp.MatchString(kv.Value.Emit())
 }
 
+// TraceFlags contains flags that can be set on a SpanContext
+type TraceFlags byte //nolint:golint
+
+// IsSampled returns if the sampling bit is set in the TraceFlags.
+func (tf TraceFlags) IsSampled() bool {
+	return tf&FlagsSampled == FlagsSampled
+}
+
+// Sampled sets the sampling bit in a new copy of the TraceFlags.
+func (tf TraceFlags) Sampled(sampled bool) TraceFlags {
```

Comment:
Do we want to name this `WithSampled` to remain consistent with the `SpanContext.With*` methods?